### PR TITLE
Feature/studentrequests

### LINKED ISSUE #41: Write post to send student date to server
- [ ] Write a post to send student data to server

### LINKED ISSUE #42: Implement React Router
- [ ] Implement React Router

### LINKED ISSUE #43: Use match.params to generate a unique link for students
- [ ] Use match.params to generate a unique link for students

### LINKED ISSUE #45: Create a mood form component
- [ ] Create a mood form component

### LINKED ISSUE #47: Write a thunk GET request, to retrieve teacher id, first and last name
- [x] Write a thunk GET request, to retrieve teacher id, students, lessons on login (or onLoad?) for teacher

### LINKED ISSUE #50: Write a thunk GET request to get the lesson for the student
- [x] Write a thunk GET request to get the lesson for the student

### LINKED ISSUE #51: Write a thunk POST to post the students mood and score
- [ ] Write a thunk POST to post the students mood and score

### PR DESCRIPTION
**Pull request by Billwilke42**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- Adds button component, add more thunk tests, adds mood POST with a score. No routing for student
## What is the new behavior?

- Button component and mood form. Tests majority of thunks, successful gets and posts for student link
## Does this introduce a breaking change?
- [ ] Yes
- [x] No

closes #50 
closes #51 
closes #41 
closes #42 
closes #43 
closes #45 
closes #47

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -34,15 +40,29 @@ function App() {
...
           <TeacherLessons />
         </Route>
-      </Switch>
-      <StudentDashboard />
+        <Route 
+          exact path='/:teacherId/:lessonId' 
+          render={({ match }) => {
+          const { teacherId, lessonId } = match.params
+          return <StudentDashboard 
+            lessonId={lessonId}
```
Comment (Melizzo) [inline src/App/App.js:48]: This is sooooo awesome Bill.  🤯

File: src/StudentDashboard/AnswerButton.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    // if(e.target.value === "false") {
+    //   setIsDisabled(true)
+    // }
+    if(e.target.value === "true") {
+      props.incrementScore(1)
+      // setIsDisabled(false)
+    }
+    if(props.lesson.questions.length === (props.currentQuestion + 1)) {
+      props.setLessonOver(true)
+    }
```
Comment (Melizzo) [inline src/StudentDashboard/AnswerButton.js:20]: This reminds me of our quiz project in the console!

File: src/App/App.js (hunk 1 above)
Comment (KyleWong2510) [reply src/App/App.js:48]: Great job generating those links!

File: src/thunks/getLessons.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { isLoading, hasErrored, setLessons } from '../actions'
 
-export const getLessons = () => {
-  const url =`https://cors-anywhere.herokuapp.com/https://gumberoo-backend.herokuapp.com/api/v1/teachers/1/lessons/`
+export const getLessons = (teacherId) => {
+  const url = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/${teacherId}/lessons`
```
Comment (Melizzo) [inline src/thunks/getLessons.js:5]: Awesome - we can now remove the cors patch!

File: src/LessonPreview/QuestionCard/QuestionCard.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ const QuestionCard = ({ id, question, allAnswers, reading, deleteQuestion})=> {
   let correctAnswer = allAnswers.find(answer => answer.correct === true).desc
   let incorrectAnswers = allAnswers
     .filter(answer => answer.correct === false)
-    .map(answer => <p className='incorrect-answers'>{answer.desc}</p>)
+    .map((answer, i )=> <p key={i++} className='incorrect-answers'>{answer.desc}</p>)
```
Comment (KyleWong2510) [inline src/LessonPreview/QuestionCard/QuestionCard.js:8]: FYI once I put my PR up, desc will have to change to the new values.  Mine should overwrite these but you might run into that elsewhere?

**Review by Melizzo — APPROVED** [review]

File: src/StudentDashboard/AnswerButton.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+})
+
+
+const mapDispatchToProps = dispatch => (
+  bindActionCreators({ incrementCurrentQuestion, setLessonOver, incrementScore }, dispatch)
+)
+
+export default connect(mapStateToProps, mapDispatchToProps)(AnswerButton)
+
+
```
Comment (KyleWong2510) [inline src/StudentDashboard/AnswerButton.js:43]: Smart to break this out into its own component for the disable.

File: src/StudentDashboard/MoodForm.js [hunk 6]
```diff
@@ -11,13 +11,31 @@ import { connect } from 'react-redux'
 function MoodForm(props) {
   //eslint-disable-next-line
   const [studentMoodInput, setStudentMoodInput] = useState('')
+  let [isSubmitted, setIsSumbitted] = useState(false)
 
+
+  const postScore = () => {
+    setIsSumbitted(true)
+    const url = `https://cors-anywhere.herokuapp.com/https://gumberoo-backend.herokuapp.com/api/v1/lessons/${props.student.id}`
```
Comment (KyleWong2510) [inline src/StudentDashboard/MoodForm.js:19]: Don't need the CORS proxy now!

File: src/StudentDashboard/MoodForm.js [hunk 7]
```diff
@@ -28,18 +46,26 @@ function MoodForm(props) {
             className='submit-mood-button' 
             aria-label='submit mood' 
             type='submit'
+            onClick={() => postScore()}
           >
             Submit
           </button>
+        </section>
+    }
+      {isSubmitted &&
```
Comment (KyleWong2510) [inline src/StudentDashboard/MoodForm.js:55]: Nice.  Much easier to read than if you had a regular ternary.

File: src/StudentDashboard/StudentForm.js [hunk 8]
```diff
@@ -1,43 +1,61 @@
...
-  props.setLesson(lesson)
+
+  const students = props.students.map((student, i)=> {
+    return (
+      <option value={student.id} key={i++}>{student.first_name}</option>
+    )
+  })
+  
+  const setStudentValues = (e) => {
+    let student = props.students.find(student => student.id === parseInt(studentNameInput))
```
Comment (KyleWong2510) [inline src/StudentDashboard/StudentForm.js:22]: Nice job with these functions!

**Review by KyleWong2510 — APPROVED** [review]
Thanks for doing those thunk tests...I hate em!

**Pull request merged**